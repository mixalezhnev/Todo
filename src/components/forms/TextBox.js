import React, { Component } from 'react';

import './TextBox.res/style.css';

export default class TextBox extends Component {
  state = { value: '' }

  handleChange = (e) => {
    this.setState({ value: e.target.value });
  }

  handleClick = (e) => {
    e.preventDefault();
    this.props.onClick(this.state.value);
  }

  render() {
    const placeholder = this.props.placeholder;

    return (
      <div className="my-textbox-component">
        <input 
          onChange={this.handleChange}
          className="textbox-input"
          type="text"
          required
          placeholder={placeholder}/>
        <button onClick={this.handleClick}>Add</button>
      </div>
    );
  }
}
