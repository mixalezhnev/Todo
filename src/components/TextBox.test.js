import React from 'react';
import { mount } from 'enzyme';

import TextBox from './TextBox';

const setup = () => {
  const props = {
    isFocused: true,
    placeholder: 'Kaaarl',
    onClick: jest.fn()
  }
  const wrapper = mount(
    <TextBox {...props} />
  );

  return {
    props,
    wrapper
  }
}

describe('components', () => {
  describe('TextBox', () => {
    it('should render self', () => {
      const { wrapper } = setup();
      expect(wrapper).toMatchSnapshot();
    });

    it('should call handleKeyPress', () => {
      const { wrapper, props } = setup();
      expect(wrapper).toMatchSnapshot();

      const input = wrapper.find('input');
      input.node.value = 'lol';
      input.simulate('keypress', {
        key: 'Enter'
      });

      expect(props.onClick).toBeCalled();
      expect(wrapper).toMatchSnapshot();
    });

    it('should call handleKeyPress with non-enter key', () => {
      const { wrapper, props } = setup();
      expect(wrapper).toMatchSnapshot();

      const input = wrapper.find('input');
      input.simulate('keypress', {
        key: 'e'
      });

      expect(props.onClick).not.toBeCalled();
      expect(wrapper).toMatchSnapshot();
    });

    it('should call handleClick', () => {
      const { wrapper, props } = setup();
      expect(wrapper).toMatchSnapshot();

      const input = wrapper.find('input');
      input.node.value = 'Vassa';

      expect(wrapper).toMatchSnapshot();

      const button = wrapper.find('button');
      button.simulate('click');

      expect(props.onClick).toBeCalled();
      expect(wrapper).toMatchSnapshot();
    });

    it('should call handleClick with empty input', () => {
      const { wrapper, props } = setup();
      expect(wrapper).toMatchSnapshot();

      const button = wrapper.find('button');
      button.simulate('click');

      expect(props.onClick).not.toBeCalled();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
