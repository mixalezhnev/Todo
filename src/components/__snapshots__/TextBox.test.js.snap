exports[`components TextBox should call handleClick 1`] = `
<TextBox
  isFocused={true}
  onClick={[Function]}
  placeholder="Kaaarl">
  <div
    className="my-textbox-component">
    <input
      autoFocus={true}
      className="textbox-input"
      onKeyPress={[Function]}
      placeholder="Kaaarl"
      type="text" />
    <button
      onClick={[Function]}>
      Add
    </button>
  </div>
</TextBox>
`;

exports[`components TextBox should call handleClick 2`] = `
<TextBox
  isFocused={true}
  onClick={[Function]}
  placeholder="Kaaarl">
  <div
    className="my-textbox-component">
    <input
      autoFocus={true}
      className="textbox-input"
      onKeyPress={[Function]}
      placeholder="Kaaarl"
      type="text" />
    <button
      onClick={[Function]}>
      Add
    </button>
  </div>
</TextBox>
`;

exports[`components TextBox should call handleClick 3`] = `
<TextBox
  isFocused={true}
  onClick={[Function]}
  placeholder="Kaaarl">
  <div
    className="my-textbox-component">
    <input
      autoFocus={true}
      className="textbox-input"
      onKeyPress={[Function]}
      placeholder="Kaaarl"
      type="text" />
    <button
      onClick={[Function]}>
      Add
    </button>
  </div>
</TextBox>
`;

exports[`components TextBox should call handleClick with empty input 1`] = `
<TextBox
  isFocused={true}
  onClick={[Function]}
  placeholder="Kaaarl">
  <div
    className="my-textbox-component">
    <input
      autoFocus={true}
      className="textbox-input"
      onKeyPress={[Function]}
      placeholder="Kaaarl"
      type="text" />
    <button
      onClick={[Function]}>
      Add
    </button>
  </div>
</TextBox>
`;

exports[`components TextBox should call handleClick with empty input 2`] = `
<TextBox
  isFocused={true}
  onClick={[Function]}
  placeholder="Kaaarl">
  <div
    className="my-textbox-component">
    <input
      autoFocus={true}
      className="textbox-input"
      onKeyPress={[Function]}
      placeholder="Kaaarl"
      type="text" />
    <button
      onClick={[Function]}>
      Add
    </button>
  </div>
</TextBox>
`;

exports[`components TextBox should call handleKeyPress 1`] = `
<TextBox
  isFocused={true}
  onClick={[Function]}
  placeholder="Kaaarl">
  <div
    className="my-textbox-component">
    <input
      autoFocus={true}
      className="textbox-input"
      onKeyPress={[Function]}
      placeholder="Kaaarl"
      type="text" />
    <button
      onClick={[Function]}>
      Add
    </button>
  </div>
</TextBox>
`;

exports[`components TextBox should call handleKeyPress 2`] = `
<TextBox
  isFocused={true}
  onClick={[Function]}
  placeholder="Kaaarl">
  <div
    className="my-textbox-component">
    <input
      autoFocus={true}
      className="textbox-input"
      onKeyPress={[Function]}
      placeholder="Kaaarl"
      type="text" />
    <button
      onClick={[Function]}>
      Add
    </button>
  </div>
</TextBox>
`;

exports[`components TextBox should call handleKeyPress with non-enter key 1`] = `
<TextBox
  isFocused={true}
  onClick={[Function]}
  placeholder="Kaaarl">
  <div
    className="my-textbox-component">
    <input
      autoFocus={true}
      className="textbox-input"
      onKeyPress={[Function]}
      placeholder="Kaaarl"
      type="text" />
    <button
      onClick={[Function]}>
      Add
    </button>
  </div>
</TextBox>
`;

exports[`components TextBox should call handleKeyPress with non-enter key 2`] = `
<TextBox
  isFocused={true}
  onClick={[Function]}
  placeholder="Kaaarl">
  <div
    className="my-textbox-component">
    <input
      autoFocus={true}
      className="textbox-input"
      onKeyPress={[Function]}
      placeholder="Kaaarl"
      type="text" />
    <button
      onClick={[Function]}>
      Add
    </button>
  </div>
</TextBox>
`;

exports[`components TextBox should render self 1`] = `
<TextBox
  isFocused={true}
  onClick={[Function]}
  placeholder="Kaaarl">
  <div
    className="my-textbox-component">
    <input
      autoFocus={true}
      className="textbox-input"
      onKeyPress={[Function]}
      placeholder="Kaaarl"
      type="text" />
    <button
      onClick={[Function]}>
      Add
    </button>
  </div>
</TextBox>
`;
